torch/nn/modules/module.py

```diff
@@ -2323,7 +2323,7 @@ def zero_grad(self, set_to_none: bool = False) -> None:
                 "The parameters are copied (in a differentiable manner) from the original module. "
                 "This means they are not leaf nodes in autograd and so don't accumulate gradients. "
                 "If you need gradients in your forward method, consider using autograd.grad instead.")
-
+        tmp_group = defaultdict(list)
```

Comment:
We might want to refactor the per_device_and_dtype_grads logic from optimizer into a helper to be able to use it here as well.

Reply:
fixed in new commit https://github.com/pytorch/pytorch/pull/91392/commits/33bda4b6c96d74724d52719920b19dbe7e8bff92